mypy/semanal.py

```diff
@@ -770,6 +770,8 @@ def analyze_base_classes(self, defn: ClassDef) -> None:
                     self.fail("Cannot subclass NewType", defn)
                 base_types.append(base)
             elif isinstance(base, AnyType):
+                if self.options.disallow_subclassing_any:
```

Comment:
I think it would be nice if this reported the name of the base class that's Any. You can probably extract that from `base_expr` at least in simple cases. Extra points if you can also report the module from it was imported (but I'm not sure if that info is easily available by the time we get here).